systemd/systemd.go

```diff
@@ -1411,56 +1410,48 @@ func ExistingMountUnitPath(mountPointDir string) string {
 
 var squashfsFsType = squashfs.FsType
 
-// Note that WantedBy=multi-user.target and Before=local-fs.target are
-// only used to allow downgrading to an older version of snapd.
-//
-// We want (see isBeforeDrivers) some snaps and components to be mounted before
-// modules are loaded (that is before systemd-{udevd,modules-load}).
-const snapMountUnitTmpl = `[Unit]
-Description={{.Description}}
+const (
+	snappyOriginModule = "X-SnapdOrigin"
+)
+
+func assembleMountUnitContent(u *MountUnitOptions) string {
+	before := "Before=snapd.mounts.target"
+	if u.MountUnitType == BeforeDriversLoadMountUnit {
+		// We want some snaps and components to be mounted before
+		// modules are loaded (that is before systemd-{udevd,modules-load}).
+		before += "\nBefore=systemd-udevd.service systemd-modules-load.service\nBefore=usr-lib-modules.mount usr-lib-firmware.mount"
+	}
+	// Note that WantedBy=multi-user.target is only used to allow
+	// downgrading to an older version of snapd.
+	unitContent := fmt.Sprintf(`[Unit]
+Description=%s
 After=snapd.mounts-pre.target
-Before=snapd.mounts.target{{if isBeforeDrivers .MountUnitType}}
-Before=systemd-udevd.service systemd-modules-load.service
-Before=usr-lib-modules.mount usr-lib-firmware.mount{{end}}
+%s
 
 [Mount]
-What={{.What}}
-Where={{.Where}}
-Type={{.Fstype}}
-Options={{join .Options ","}}
+What=%s
```

Comment:
For big format string, I would use the index, e.g. `%[1]s`. So that we do not have to look at the whole string and count.

Reply:
Good point, done